File: src/main/java/me/geek/mail/common/DataBase/DataSub.java
package me.geek.mail.common.DataBase;

import java.sql.Connection;
import java.sql.SQLException;

/**
 * 作者: 老廖
 * 时间: 2022/7/23
 **/
public abstract class DataSub {
    public abstract Connection getConnection() throws SQLException;
    public abstract void onLoad();
    public abstract void onStop();
}

File: src/main/kotlin/me/geek/mail/common/data/sub/DataSub.kt
package me.geek.mail.common.data.sub

import java.sql.Connection

/**
 * 作者: 老廖
 * 时间: 2022/7/23
 */
abstract class DataSub {

    abstract val connection: Connection

    abstract fun onLoad()

    abstract fun onStop()
}